### PR TITLE
hex_a_hop: Add recipe

### PR DESCRIPTION
**Pull request by win8linux**
*Hex-a-Hop* is a hexagonal tile-based puzzle game wherein the player must eliminate all green tiles by walking on them. Limitless undos, no time limits, just get rid of the green tiles!  Different kinds of tiles are introduced as the player progresses further for variety.

Running on `x86_64`:
screenshot1
screenshot3
screenshot5
screenshot6

### CODE REVIEW

**Comment by win8linux** [comment]
This is the first time I've encountered Autotools, so there may be some unnecessary build steps that need to be removed.

File: games-puzzle/hex_a_hop/hex_a_hop-1.1.0.recipe [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+COPYRIGHT="2005-2007 Tom Beaumont"
+LICENSE="GNU GPL v2"
+REVISION="1"
+SOURCE_URI="https://downloads.sourceforge.net/project/hexahop/$portVersion/hex-a-hop-$portVersion.tar.gz"
+CHECKSUM_SHA256="80bf19d936a8430cab5bc468ee6827c38052e026c609eb7ece16317e7932a0d6"
+SOURCE_DIR="hex-a-hop-$portVersion"
+PATCHES="hex_a_hop-$portVersion.patchset"
+
+ARCHITECTURES="all !x86_gcc2"
+SECONDARY_ARCHITECTURES="!x86"
```
Comment (Begasus) [inline games-puzzle/hex_a_hop/hex_a_hop-1.1.0.recipe:21]: OK for secondary architecture: 
```diff
-SECONDARY_ARCHITECTURES="!x86"
+SECONDARY_ARCHITECTURES="x86"
```

File: games-puzzle/hex_a_hop/hex_a_hop-1.1.0.recipe [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+	autoreconf -vfi
+	automake --add-missing
+	runConfigure --omit-dirs "binDir dataRootDir" \
+		./configure --bindir=$appsDir
+	make $jobArgs
+}
+
+INSTALL()
+{
+	mkdir -p $appsDir
```
Comment (Begasus) [inline games-puzzle/hex_a_hop/hex_a_hop-1.1.0.recipe:63]: Not needed:
```diff
-	mkdir -p $appsDir
```

**Review by Begasus — CHANGES_REQUESTED** [review]

**Review by win8linux — COMMENTED** [review]
All requested changes have been made.

File: games-puzzle/hex_a_hop/hex_a_hop-1.1.0.recipe [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+DESCRIPTION="Hex-a-Hop is a hexagonal tile-based puzzle game with one simple \
+goal: destroy all green tiles! There are infinite undos and no time limits -- \
+you just have to find a way to destroy all the green tiles and step on a safe \
+tile at the end.
+
+As you progress through the game, more types of tiles are introduced which \
+make things more difficult and interesting as you progress through the many \
+levels. You can use the infinite undo feature to retrace your steps if you \
+make a mistake."
+HOMEPAGE="http://hexahop.sourceforge.net"
```
Comment (humdingerb) [inline games-puzzle/hex_a_hop/hex_a_hop-1.1.0.recipe:11]:
```suggestion
HOMEPAGE="https://hexahop.sourceforge.net"
```

File: games-puzzle/hex_a_hop/hex_a_hop-1.1.0.recipe [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+	runConfigure --omit-dirs "binDir dataRootDir" \
+		./configure --bindir=$appsDir
+	make $jobArgs
+}
+
+INSTALL()
+{
+	make install
+
+	addAppDeskbarSymlink $appsDir/hex-a-hop "Hex-a-hop"
```
Comment (Begasus) [inline games-puzzle/hex_a_hop/hex_a_hop-1.1.0.recipe:65]: Nitpicking maybe :) (suggestion)
```
	mv $appsDir/hex-a-hop $appsDir/Hex-a-hop
	addAppDeskbarSymlink $appsDir/Hex-a-hop
```

**Review by Begasus — APPROVED** [review]

**Comment by Begasus** [comment]
@humdingerb maybe you could bring your magic skills and create an icon? ;)

**Comment by Begasus** [comment]
LGTM, if only we had an icon, but can be added later. Thanks @win8linux

**Review by humdingerb — APPROVED** [review]

**Comment by humdingerb** [comment]
> @humdingerb maybe you could bring your magic skills and create an icon? ;)

I'm not that good an iconizer... we'd need @Handmaus for best results. 😃

**Comment by win8linux** [comment]
Could just make an issue for an icon after the merge.

**Comment by humdingerb** [comment]
> Could just make an issue for an icon after the merge.

Yes, and/or advertise on this thread:
https://discuss.haiku-os.org/t/missing-icons-rdef-hvif-iom

**Comment by win8linux** [comment]
> Yes, and/or advertise on this thread:
> https://discuss.haiku-os.org/t/missing-icons-rdef-hvif-iom

Right, make the issue then post it on that thread. Got it.

**Pull request merged**